src/olympia/promoted/models.py

```diff
@@ -0,0 +1,57 @@
+from django.db import models
+
+from olympia.addons.models import Addon
+from olympia.amo.models import ModelBase
+from olympia.constants.promoted import (
+    NOT_PROMOTED, PROMOTED_GROUPS, PROMOTED_GROUPS_BY_ID)
+from olympia.versions.models import Version
+
+
+class PromotedAddon(ModelBase):
+    GROUP_CHOICES = [(group.id, group.name) for group in PROMOTED_GROUPS]
+    group_id = models.SmallIntegerField(
+        choices=GROUP_CHOICES, default=NOT_PROMOTED.id)
+    addon = models.OneToOneField(
+        Addon, on_delete=models.CASCADE, null=False)
+
+    @property
+    def group(self):
+        return PROMOTED_GROUPS_BY_ID.get(self.group_id, NOT_PROMOTED)
+
+    def __getattr__(self, name):
+        # magic to return the fixed properties of the promoted group too
+        try:
+            super().__getattribute__(name)
+        except AttributeError as parent_attr_error:
+            try:
+                return getattr(self.group, name)
+            except AttributeError:
+                raise parent_attr_error
+
+    def is_addon_promoted(self, addon=None):
+        """Is the addon approved for promotion within the *current* promoted
+        group."""
+        addon = addon or self.addon  # the caller might already have the addon
+        if not addon or not addon.current_version:
+            return False
+        return bool(
+            self.group != NOT_PROMOTED.id and
+            addon.current_version.promoted_approvals.filter(
```

Comment:
So just the existence of a `PromotedApproval` means that the `addon` was approved for this group - there isn't a boolean `approved` property?

Reply:
Correct - the record would only be created on approval.  Strategic group promoted addons are a ❔ right now because they're auto-approved - I guess we'd just created it on auto approval.